### PR TITLE
osparc-service cookiecutter updates to run under gitlab-ci

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- [x] change user ID on files created before entrypoint.sh is run
- [x] general refactoring
- [x] tests run under gitlab now (updated integration stub)

Note: still not sure about security of having docker in docker in a CI env.

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] **Runs in the swarm**
- [x] If you design a new module, add your user to .github/CODEOWNERS

[waffle.io]:https://waffle.io/marketing-assets/documents/waffleio_cheatsheet_v1.pdf?utm_source=blog&utm_medium=cheatsheet-ctabutton&utm_campaign=cheatsheet

### CODE REVIEW

File: {{cookiecutter.project_slug}}/tools/update_compose_labels.py [hunk 1]
```diff
@@ -1,75 +1,74 @@
...
-
-
-if __name__ == "__main__":
-    main()
-  
\ No newline at end of file
+#!/bin/python
+
+###
+# Usage: python update_compose_labels --c docker-compose.yml -f folder/path
```
Comment (pcrespov) [inline {{cookiecutter.project_slug}}/tools/update_compose_labels.py]: Reuse the file header text as help in command line. For that you have to write it with triple quote and then retrieve it using ``__doc__``. For instance:
```python
""" This is my help ...

"""

...

parser = argparse.ArgumentParser( description=" bla bla \n" + __doc__ )
```

File: {{cookiecutter.project_slug}}/tools/run_creator.py [hunk 2]
```diff
@@ -39,8 +39,8 @@ def get_output_config(folder: Path) -> Dict:
     for input_key, input_value in input_config["inputs"].items():
```
Comment (pcrespov) [inline {{cookiecutter.project_slug}}/tools/run_creator.py]: I would always have a minimal ``if __name__=="__main__"`` section and create a ``main`` function. IMHO it is as a very good practice that makes that part of the code testable, reusable and composable (see other comments)

File: {{cookiecutter.project_slug}}/docker/entrypoint.sh [hunk 3]
```diff
@@ -11,43 +11,67 @@ echo "  User    :`id $(whoami)`"
...
+        (echo "ERROR: You must mount '$INPUT_FOLDER' to deduce user and group ids" && exit 1)
+stat $OUTPUT_FOLDER &> /dev/null || \
+    (echo "ERROR: You must mount '$OUTPUT_FOLDER' to deduce user and group ids" && exit 1)
+stat $LOG_FOLDER &> /dev/null || \
+    (echo "ERROR: You must mount '$LOG_FOLDER' to deduce user and group ids" && exit 1)
 
-    stat $DEVEL_MOUNT &> /dev/null || \
-        (echo "ERROR: You must mount '$DEVEL_MOUNT' to deduce user and group ids" && exit 1) # FIXME: exit does not stop script
+stat $INPUT_FOLDER &> /dev/null
+if [[ $? -eq 0 ]]
```
Comment (pcrespov) [inline {{cookiecutter.project_slug}}/docker/entrypoint.sh]: no need for this check. Redundant

File: {{cookiecutter.project_slug}}/docker/entrypoint.sh [hunk 4]
```diff
@@ -11,43 +11,67 @@ echo "  User    :`id $(whoami)`"
...
 
-    USERID=$(stat -c %u $DEVEL_MOUNT)
-    GROUPID=$(stat -c %g $DEVEL_MOUNT)
+    USERID=$(stat -c %u $INPUT_FOLDER)
+    GROUPID=$(stat -c %g $INPUT_FOLDER)
     GROUPNAME=$(getent group ${GROUPID} | cut -d: -f1)
-
     if [[ $USERID -eq 0 ]]
     then
         addgroup scu root
```
Comment (pcrespov) [inline {{cookiecutter.project_slug}}/docker/entrypoint.sh]: I would parametrize scu as well. Perhaps we should add ``SC_USER_NAME`` variable in ``Dockerfile``.

**Comment by pcrespov** [comment]
We should also try to find a better naming for these repos ... it is easy to confuse with ``simcore-service``. But we do not do it until we have a glossary with definitions and proper lexicon

File: {{cookiecutter.project_slug}}/tools/update_compose_labels.py [hunk 5]
```diff
@@ -1,75 +1,74 @@
...
+        changed = True
+    return changed
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Update a docker-compose file with json files in a path")
+    parser.add_argument(
+        "--compose", help="The compose file where labels shall be updated", type=Path, required=True)
+    parser.add_argument("--input", help="The json folder to stringify", type=Path, required=True)
+    args = sys.argv[1:]
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply {{cookiecutter.project_slug}}/tools/update_compose_labels.py]: done

File: {{cookiecutter.project_slug}}/tools/update_compose_labels.py [hunk 6]
```diff
@@ -1,75 +1,74 @@
...
+            log.info("Updating %s using labels in %s", options.compose, options.input)
+            # write the file back
+            with options.compose.open('w') as fp:
+                yaml.safe_dump(compose_cfg, fp, default_flow_style=False)
+                log.info("Update completed")
+        else:
+            log.info("No update necessary")
+    except: #pylint: disable=bare-except
+        log.exception("Unexpected error:")
+        sys.exit(1)
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply {{cookiecutter.project_slug}}/tools/update_compose_labels.py]: used IntEnum and returns int from main. Done.

File: {{cookiecutter.project_slug}}/tools/update_compose_labels.py (hunk 1 above)
Comment (sanderegg) [reply {{cookiecutter.project_slug}}/tools/update_compose_labels.py]: did not know that. Done.

File: {{cookiecutter.project_slug}}/tools/run_creator.py (hunk 2 above)
Comment (sanderegg) [reply {{cookiecutter.project_slug}}/tools/run_creator.py]: you're right. Done and added also sys.exit etc.

File: {{cookiecutter.project_slug}}/docker/entrypoint.sh (hunk 3 above)
Comment (sanderegg) [reply {{cookiecutter.project_slug}}/docker/entrypoint.sh]: done.

File: {{cookiecutter.project_slug}}/docker/entrypoint.sh (hunk 4 above)
Comment (sanderegg) [reply {{cookiecutter.project_slug}}/docker/entrypoint.sh]: done.

**Pull request merged**